website/docs/advanced-plugins.md

```diff
@@ -28,9 +38,75 @@ Plugins are modules which export a function that takes in the context, options a
 
 -->
 
+## Creating plugins
+
+A plugin is a module which exports a function that takes two parameters and returns an object.
+
+We need to specify for our plugin:
+
+- which files to watch
+- generate which pages from those files
+- in those pages, which components to call and with what props
+
+We'll use [@docusaurus/plugin-content-blog](https://github.com/facebook/docusaurus/tree/master/packages/docusaurus-plugin-content-blog) as an example to explain how to create a plugin.
+
+### Module definition
+
+The exported modules for plugins are called with two parameters: `context` and `options`:
+
+<!-- packages/docusaurus/src/server/types.ts -->
+
+```js
+module.exports = function(context, options) {
+  // ...
+};
+```
+
+`context` contains the following fields:
+
+```js
+interface LoadContext {
+  siteDir: string;
+  generatedFilesDir: string;
+  siteConfig: DocusaurusConfig;
+  cliOptions: CLIOptions;
+  outDir: string;
+  baseUrl: string;
+}
+
+interface CLIOptions {
+  [option: string]: any;
+}
+```
+
+And `options` are optionally the [second parameter when the plugins are used](/docs/using-plugins#configuring-plugins).
+
+### Paths to watch
```

Comment:
I wonder why we only mention paths to watch here? Is this supposed to be a tutorial or some sort?